tests/integ/modin/binary/test_binary_op.py

```diff
@@ -1299,10 +1298,20 @@ def check_op(native_lhs, native_rhs, snow_lhs, snow_rhs):
             snow_ans, native_ans, lambda s: s, check_index_type=False
         )
 
-    check_op(lhs, rhs, try_cast_to_snow_series(lhs), try_cast_to_snow_series(rhs))
+    # The join count is high because:
+    # - When creating a single index Series, 1 join is performed; four series are created.
```

Comment:
why 1 join here? I believe here pd.Series get both data and index in local list. there should not have any join though.

Reply:
`set_index` is used to set the index even if it's a local list - previously `from_pandas` was used to obtain a native pandas dataframe (which would automatically set the index). But doing it that way prevents us from covering cases where a Series/DataFrame data is provided, and an index/columns is provided where some elements in the index/columns don't exist in the Series/DataFrame.

Reply:
sometimes using `from_pandas(native pandas df)` can give us an incorrect intermediate result in the constructor, so I avoid doing that entirely and we handle setting the index

Reply:
I think we have discussed this before, if all inputs are local, we should use `from_pandas` to avoid joins. What was the issue here?

Reply:
I can make it so that if all the data given is local, avoid joins. But if data or index is a snowpark pandas object there's going to be a join

Reply:
Yes we should avoid joins if all inputs are local. If any input is modin objects, we cannot avoid join. This case should be the former one.